### ISSUE TITLE
Error: GSettings key thermal-sensor-label not found in schema org.gnome.shell.extensions.system-monitor

### ISSUE THREAD

**Issue by grahamwhiteuk** [mgalgs/gnome-shell-system-monitor-applet#91]
The extension stopped working for me today with the following message found in the gnome extensions app:

> An error has occurred in this extensions.  This could cause issues elsewhere in the system.  It is recommended to turn the extension off until this error is resolved.
> 
> Error details:
> 
> Error: GSettings key thermal-sensor-label not found in schema org.gnome.shell.extensions.system-monitor

There's a stack trace in the journal log as follows:

```
Extension system-monitor-next@paradoxxx.zero.gmail.com: Error: GSettings key thermal-sensor-label not found in schema org.gnome.shell.extensions.system-monitor
                                   
                                   Stack trace:
                                     _checkKey@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:779:23
                                     createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:729:30
                                     refresh@file:///usr/share/gnome-shell/extensions/system-monitor-next@paradoxxx.zero.gmail.com/extension.js:2021:44
                                     update@file:///usr/share/gnome-shell/extensions/system-monitor-next@paradoxxx.zero.gmail.com/extension.js:954:14
                                     SystemMonitor_Thermal@file:///usr/share/gnome-shell/extensions/system-monitor-next@paradoxxx.zero.gmail.com/extension.js:2018:14
                                     enable@file:///usr/share/gnome-shell/extensions/system-monitor-next@paradoxxx.zero.gmail.com/extension.js:2363:66
                                     _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:266:38
                                     loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:478:32
                                     async*_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:786:24
                                     async*_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:792:48
                                     _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:827:20
                                     async*init@resource:///org/gnome/shell/ui/extensionSystem.js:76:14
                                     _initializeUI@resource:///org/gnome/shell/ui/main.js:303:22
                                     start@resource:///org/gnome/shell/ui/main.js:175:11
                                     @resource:///org/gnome/shell/ui/init.js:12:47
                                     @resource:///org/gnome/shell/ui/init.js:21:20
```

I'm running on Fedora 40 with all the latest updates applied.  My hardware is a Lenovo ThinkPad P15 Gen 1 laptop.

**Comment by grahamwhiteuk** [comment]
I forgot to mention which version of the extension I'm running.  I'm currently using the one installed from RPM which is `gnome-shell-extension-system-monitor-applet-38-30.20240717git4bd03a2.fc40.noarch`

**Comment by mgalgs** [comment]
Interesting, can you please post the output of `dconf list /org/gnome/shell/extensions/system-monitor/`?

**Comment by grahamwhiteuk** [comment]
Sure...

```shell
$ dconf list /org/gnome/shell/extensions/system-monitor/
center-display
compact-display
disk-display
disk-show-menu
disk-show-text
fan-display
gpu-display
icon-display
show-tooltip
```

**Comment by grahamwhiteuk** [comment]
hmm, I just applied the latest updates from Fedora and I can no longer reproduce this issue.  Not sure if you want to close this one off but if there's anything further I can do to help diagnose/debug then let me know.

**Comment by mgalgs** [comment]
All right, my initial guess here is that maybe you had the og version of the extension installed ([paradoxxxzero/gnome-shell-system-monitor-applet](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet)) and then uninstalled that and installed this version, which expects some different settings keys. I thought the uninstall process nuked the dconf entries so I'm not sure how that would actually happen. It might be related to the fact that our schema ID currently matches the og version (which we're trying to rectify in #65), combined with some kind of incomplete uninstall or something...

We should probably handle busted configs more gracefully but it might be a rare enough case to not be worth the effort... I'll leave this open to see if we get any more bites on this error message.

**Comment by mgalgs** [comment]
I believe that this will be fixed by #98, since that will solve the overlapping schema ID issue.
